pkg/publisher/publisher.go

```diff
@@ -85,21 +85,33 @@ type cloudWatchPublisher struct {
 }
 
 // New returns a new instance of `Publisher`
-func New(ctx context.Context) (Publisher, error) {
+func New(ctx context.Context, region string, clusterID string) (Publisher, error) {
 	sess := awssession.New()
-	// Get cluster-ID
-	ec2Client, err := ec2wrapper.NewMetricsClient()
-	if err != nil {
-		return nil, errors.Wrap(err, "publisher: unable to obtain EC2 service client")
-	}
-	clusterID := getClusterID(ec2Client)
 
-	// Get ec2metadata client
-	ec2MetadataClient := ec2metadatawrapper.New(sess)
+	// Case 1: Cx not using IRSA, we need to get region and clusterID using IMDS (existing approach)
+	// Case 2: Cx using IRSA but not specified clusterID, we can still get this info if IMDS is not blocked
+	// Case 3: Cx blocked IMDS access and not using IRSA (which means region == "") OR
+	// not specified clusterID then its a Cx error
+	if region == "" || clusterID == "" {
+		ec2Client, err := ec2wrapper.NewMetricsClient()
+		if err != nil {
+			return nil, errors.Wrap(err, "publisher: unable to obtain EC2 service client")
+		}
 
-	region, err := ec2MetadataClient.Region()
-	if err != nil {
-		return nil, errors.Wrap(err, "publisher: Unable to obtain region")
+		// If Customers have explicitly specified clusterID then skip auto generating it
+		if clusterID == "" {
+			clusterID = getClusterID(ec2Client)
+		}
+
+		// Get ec2metadata client
+		ec2MetadataClient := ec2metadatawrapper.New(sess)
```

Comment:
Why do this outside of the below `if` loop? We don't need to do this if `region` is already available isn't it